tss-esapi/src/structures/buffers/public/ecc.rs

```diff
@@ -0,0 +1,270 @@
+            is_signing_key: false,
+            is_decryption_key: true,
+            restricted: true,
+        }
+    }
+
+    /// Create parameters for an unrestricted signing key
+    pub fn new_unrestricted_signing_key(scheme: EccScheme, curve: EccCurve) -> Self {
+        PublicEccParametersBuilder {
+            symmetric: None,
+            ecc_scheme: Some(scheme),
+            ecc_curve: Some(curve),
+            key_derivation_function_scheme: Some(KeyDerivationFunctionScheme::Null),
+            is_signing_key: true,
+            is_decryption_key: false,
+            restricted: false,
+        }
+    }
+
+    /// Adds a [SymmetricDefinitionObject] to the [PublicEccParametersBuilder].
+    pub const fn with_symmetric(mut self, symmetric: SymmetricDefinitionObject) -> Self {
+        self.symmetric = Some(symmetric);
+        self
+    }
+
+    /// Adds a [EccScheme] to the [PublicEccParametersBuilder].
+    pub const fn with_ecc_scheme(mut self, ecc_scheme: EccScheme) -> Self {
+        self.ecc_scheme = Some(ecc_scheme);
+        self
+    }
+
+    /// Adds [EccCurve] to the [PublicEccParametersBuilder].
+    pub const fn with_curve(mut self, ecc_curve: EccCurve) -> Self {
+        self.ecc_curve = Some(ecc_curve);
+        self
+    }
+
+    /// Adds [KeyDerivationFunctionScheme] to the [PublicEccParametersBuilder].
+    pub const fn with_key_derivation_function_scheme(
+        mut self,
+        key_derivation_function_scheme: KeyDerivationFunctionScheme,
+    ) -> Self {
+        self.key_derivation_function_scheme = Some(key_derivation_function_scheme);
+        self
+    }
+
+    /// Adds a flag that indicates if the key is going to be used
+    /// for signing to the [PublicEccParametersBuilder].
+    ///
+    /// # Arguments
+    /// * `set` - `true` inidcates that the key is going to be used for signing operations.
+    ///           `false` indicates that the key is not going to be used for signing operations.
+    pub const fn with_is_signing_key(mut self, set: bool) -> Self {
+        self.is_signing_key = set;
+        self
+    }
+
+    /// Adds a flag that indicates if the key is going to be used for
+    /// decryption to the [PublicEccParametersBuilder].
+    ///
+    /// # Arguments
+    /// * `set` - `true` indicates that the key is going to be used for decryption operations.
+    ///           `false` indicates that the key is not going to be used for decryption operations.
+    pub const fn with_is_decryption_key(mut self, set: bool) -> Self {
+        self.is_decryption_key = set;
+        self
+    }
+
+    /// Adds a flag that inidcates if the key is going to be restrictied to
+    /// the [PublicEccParametersBuilder].
+    ///
+    /// # Arguments
+    /// * `set` - `true` indicates that it is going to be a restricted key.
+    ///           `false` indicates that it is going to be a non restricted key.
+    pub const fn with_restricted(mut self, set: bool) -> Self {
```

Comment:
Shouldn't this be `with_is_restricted` to mirror `with_is_signing_key` and `with_is_decryption_key`? :thinking: